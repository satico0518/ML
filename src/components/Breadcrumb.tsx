import React from 'react';

export const Breadcrumb = () => {
  return (
    <ul className="breadcrumb">
      <li>
        <a href="!#">Electronica, Audio y Video</a> >
      </li>
      <li>
        <a href="!#">iPod</a> >
      </li>
      <li>
        <a href="!#">Reproductores</a> >
      </li>
      <li>
        <a href="!#">Reproductores</a> >
      </li>
      <li>
        <a href="!#">iPod touch</a> >
      </li>
      <li>
        <a href="!#">32 GB</a>
      </li>
    </ul>
  );
};
