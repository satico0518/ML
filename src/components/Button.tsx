import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ text }: { text: string }) => <div className='custom-button'>{text}</div>;

Button.propTypes = {
  text: PropTypes.string.isRequired,
};

export default Button;
