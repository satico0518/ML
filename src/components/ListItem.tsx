import React, { FC } from 'react';
import PropTypes from 'prop-types';
import { Item } from '../types';
import { Link } from 'react-router-dom';

const ListItem: FC<Item> = ({ id, price, title, state_name, thumbnail }) => {
  return (
    <article>
      <figure>
        <Link to={`/items/${id}`}>
          <img src={thumbnail} alt={title} />
        </Link>
      </figure>
      <aside>
        <div>
          <h2>{price}</h2>
          <p>{title}</p>
        </div>
        <div>{state_name}</div>
      </aside>
    </article>
  );
};

ListItem.propTypes = {
  price: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  state_name: PropTypes.string.isRequired,
};

export default ListItem;
