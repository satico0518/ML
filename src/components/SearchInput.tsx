import React, { FC, ChangeEvent } from 'react';
import PropTypes from 'prop-types';

interface SearchInputProps {
  placeholder: string;
  query: string;
  setQuery: Function
}

const SearchInput: FC<SearchInputProps> = ({ placeholder, query, setQuery }) => {
  const handleQueryChange = (e: ChangeEvent<HTMLInputElement>) =>
    setQuery(e.target.value);

  return (
    <input className='search-input'
      type="search"
      name="query"
      value={query}
      placeholder={placeholder}
      onChange={handleQueryChange}
    />
  );
};

SearchInput.propTypes = {
  placeholder: PropTypes.string.isRequired,
  query: PropTypes.string.isRequired,
  setQuery: PropTypes.func.isRequired
};

export default SearchInput;
