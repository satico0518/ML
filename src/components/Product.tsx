import React from 'react';
// redux
import { useSelector } from 'react-redux';
// types
import { AppState } from '../redux/types';
// helpers
import { currencyFormat, translateCondition } from '../utils';
// components
import Button from './Button';

const Product = () => {
  const { selectedItem } = useSelector((state: AppState) => state.items);
  const currentItem = selectedItem?.item;

  return (
    <main className="product__wrapper">
      <section>
        <figure>
          <img src={currentItem?.picture} alt={currentItem?.title} />
        </figure>
        <aside>
          <span>
            {translateCondition(currentItem?.condition || '')} -{' '}
            {currentItem?.sold_quantity || 0} vendidos
          </span>
          <span className='product__title'>{currentItem?.title}</span>
          <span className='product__price'>{currencyFormat(currentItem?.price.amount || 0)}</span>
          <span style={{ marginRight: 32 }}><Button text="Comprar"/></span>
        </aside>
      </section>
      <footer>
        <strong className="product__description-title">
          Descripci√≥n del Producto
        </strong>
        <article className="product__description-text">
          {currentItem?.description}
        </article>
      </footer>
    </main>
  );
};

export default Product;
