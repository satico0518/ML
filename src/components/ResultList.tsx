import React, { FC } from 'react';
// components
import ListItem from './ListItem';
// types
import { Item } from '../types';

interface Props {
  items: Item[];
}

const ResultList: FC<Props> = ({ items }) => {
  return (
    <section className="result-list">
      {items?.map((item) => (
        <ListItem
          key={item.id}
          id={item.id}
          price={item.price}
          title={item.title}
          picture={item.picture}
          state_name={item.state_name}
          condition={item.condition}
          free_shipping={item.free_shipping}
        />
      ))}
    </section>
  );
};

export default ResultList;
