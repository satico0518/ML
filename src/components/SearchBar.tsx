import React, { useState, FormEvent, useEffect } from 'react';
// componests
import SearchInput from './SearchInput';
// assets
import logo from '../assets/images/Logo_ML.png';
import searchIcon from '../assets/images/ic_Search.png';
// hooks
import { useHistory } from 'react-router-dom';
import { useSelector } from 'react-redux';
// types
import { AppState } from '../redux/types';

const SearchBar = () => {
  const { query: queryword } = useSelector((state: AppState) => state.items);
  const history = useHistory();
  const [query, setQuery] = useState('');
  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    if (!query.length) return;
    history.push(`/items?search=${query}`);
  };
  useEffect(() => {
    setQuery(queryword);
  }, [queryword]);
  return (
    <header className="search-bar__wrapper">
      <img className="search-bar__img" src={logo} alt="logo_ML" />
      <form className="search-bar__form" onSubmit={handleSubmit}>
        <SearchInput
          placeholder="Nunca dejes de buscar"
          query={query}
          setQuery={setQuery}
        />
        <button type="submit" title="Search this website now">
          <img src={searchIcon} alt="search icon" />
        </button>
      </form>
    </header>
  );
};

export default SearchBar;
