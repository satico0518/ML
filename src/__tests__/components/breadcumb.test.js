import React, { Component } from 'react';

import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { Breadcrumb } from '../../components/Breadcrumb';

describe('breadcumb', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(<Breadcrumb />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should search if link is clicked', () => {
    const wrapper = shallow(<Breadcrumb />);
    const firstLink = wrapper.find('Link').first();
    const toProp = firstLink.prop('to');
    const text = firstLink.text();
    expect(toProp).toBe(`/items?search=${text}`);
  });
});
