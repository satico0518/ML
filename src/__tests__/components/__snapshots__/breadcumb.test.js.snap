// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breadcumb should match snapshot 1`] = `
<ul
  className="breadcrumb"
>
  <li
    key="Electronica, Audio y Video"
  >
    <Link
      to="/items?search=Electronica, Audio y Video"
    >
      Electronica, Audio y Video
    </Link>
     
     &gt;
  </li>
  <li
    key="iPod"
  >
    <Link
      to="/items?search=iPod"
    >
      iPod
    </Link>
     
     &gt;
  </li>
  <li
    key="Reproductores"
  >
    <Link
      to="/items?search=Reproductores"
    >
      Reproductores
    </Link>
     
     &gt;
  </li>
  <li
    key="iPod touch"
  >
    <Link
      to="/items?search=iPod touch"
    >
      iPod touch
    </Link>
     
     &gt;
  </li>
  <li
    key="32 GB"
  >
    <Link
      to="/items?search=32 GB"
    >
      32 GB
    </Link>
     
  </li>
</ul>
`;
